src/mantle/testing/concerns/trait-assertions.php

```diff
@@ -368,4 +371,82 @@ public function assertUserDoesNotExists( array $arguments ) {
 
 		PHPUnit::assertEmpty( $users );
 	}
+
+	/**
+	 * Get a term object from a flexible argument.
+	 *
+	 * @param Term|\WP_Term|int $argument Term object, term ID, or term slug.
+	 * @return WP_Term|null
+	 */
+	protected function get_term_from_argument( $argument ): ?WP_Term {
+		if ( $argument instanceof Term ) {
+			return $argument->core_object();
+		}
+
+		if ( is_int( $argument ) ) {
+			return get_term_object( $argument );
+		}
+
+		if ( $argument instanceof WP_Term ) {
+			return $argument;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Assert that a post has a specific term.
+	 *
+	 * `assertPostNotHasTerm()` is the inverse of this method.
+	 *
+	 * @param Post|\WP_Post|int $post Post to check.
+	 * @param Term|\WP_Term|int $term Term to check.
+	 * @return void
+	 */
+	public function assertPostHasTerm( $post, $term ) {
+		if ( $post instanceof Post ) {
+			$post = $post->id();
+		}
+
+		$term = $this->get_term_from_argument( $term );
+
+		if ( $term ) {
+			PHPUnit::assertTrue( \has_term( $term->term_id, $term->taxonomy, $post ) );
+		} else {
+			PHPUnit::fail( 'Term not found to assert against' );
+		}
+	}
+
+	/**
+	 * Assert that a post doesn't have a specific term.
+	 *
+	 * `assertPostHasTerm()` is the inverse of this method.
+	 *
+	 * @param Post|\WP_Post|int $post Post to check.
+	 * @param Term|\WP_Term|int $term Term to check.
+	 * @return void
+	 */
+	public function assertPostNotHasTerm( $post, $term ) {
+		if ( $post instanceof Post ) {
+			$post = $post->id();
+		}
+
+		$term = $this->get_term_from_argument( $term );
+
+		if ( $term ) {
+			PHPUnit::assertFalse( \has_term( $term->term_id, $term->taxonomy, $post ) );
+		} else {
+			PHPUnit::fail( 'Term not found to assert against' );
+		}
+	}
```

Comment:
I think the logic is off on this one a bit. We're verifying that a term doesn't exist, so we shouldn't fail if `$term` is empty.

```suggestion
	public function assertPostNotHasTerm( $post, $term ) {
		if ( $post instanceof Post ) {
			$post = $post->id();
		}

		$term = $this->get_term_from_argument( $term );
		
		PHPUnit::assertNotInstanceOf( \WP_Term::class, $term, 'Term found to assert against' );
		PHPUnit::assertFalse( \has_term( $term->term_id, $term->taxonomy, $post ) );
	}
```